lib/fhir_models/bootstrap/generator.rb

```diff
@@ -174,8 +173,17 @@ def generate_class(hierarchy, structure_def, top_level = false)
             end
 
             # generate a field for each valid datatype... this is for things like Resource.attribute[x]
-            element['type'].map { |t| t['code'] }.uniq.each do |data_type|
+            element['type'].each do |type|
+              extension = type['extension']
+              data_type = type['code']
               data_type = 'string' unless data_type
+              if element['path'].end_with?('.id')
+                data_type = element['base']['path'] == 'Resource.id' ? 'id' : 'string'
+              end
+              if data_type == 'http://hl7.org/fhirpath/System.String' && extension
```

Comment:
looks like this could be an `elsif` based on the preceding if statement

Reply:
No, they are not mutually exclusive conditions.

Reply:
it looks like the first if statement will set `data_type` to either `id` or `string`.

if that happens then the second if statment condition will always be false `data_type == http://hl7.org/fhirpath/System.String`.

Am I misreading something here and making a fool of myself?

Reply:
It doesn't always execute.

Reply:
Or maybe I am making a fool of myself. Whatever. It works, that is the important thing.

Reply:
There is no fool emoji, apparently, which is tragic.

Reply:
`:black_joker:` 🃏 may be the best match, but apparently it can't be used as a reaction.

Reply:
But whenever the first if does execute, the second one never will.  So we could do this below and avoid checking the second condition if the first evaluates to true.
```ruby
if element['path'].end_with?('.id')
  data_type = element['base']['path'] == 'Resource.id' ? 'id' : 'string'
elsif data_type == 'http://hl7.org/fhirpath/System.String' && extension
  data_type = extension.first['valueUrl']
end
```

It's not a big deal, so if you prefer it the way it is that's fine with me.

Reply:
you have my approval either way.

Reply:
Fixed, only because you are so passionate about it.